superset-frontend/src/explore/components/ControlPanelsContainer.jsx

```diff
@@ -92,6 +92,27 @@ class ControlPanelsContainer extends React.Component {
     this.renderControlPanelSection = this.renderControlPanelSection.bind(this);
   }
 
+  componentDidUpdate(prevProps) {
+    const {
+      actions: { setControlValue },
+    } = this.props;
+    if (this.props.form_data.datasource !== prevProps.form_data.datasource) {
+      const defaultValues = [
+        'MetricsControl',
+        'AdhocFilterControl',
+        'TextControl',
+        'SelectControl',
+        'CheckboxControl',
+      ];
```

Comment:
Probably not a big concern, but `AnnotationLayerControl` would probably be good to include here.